lib/blueprinter/view.rb

```diff
@@ -1,14 +1,22 @@
 module Blueprinter
   # @api private
   class View
-    attr_reader :excluded_field_names, :fields, :included_view_names, :name, :transformers
+    attr_reader :excluded_field_names, :fields, :included_view_names, :name, :transformers, :definition_order
 
     def initialize(name, fields: {}, included_view_names: [], excluded_view_names: [],transformers: [])
       @name = name
       @fields = fields
       @included_view_names = included_view_names
       @excluded_field_names = excluded_view_names
       @transformers =  transformers
+      @definition_order = []
```

Comment:
Structs with a name and a type (```:field``` or ```:view```) each? The main thing is to avoid the possibility of name collisions between views and fields without hacky magic substrings in the array like ```"actually_its_a_view_#{view_name}"``` . ```include_view``` support means we can't just stick the view object itself in there.

Reply:
That would work!

Reply:
As the last push it is an array of ```DefinitionPlaceholder``` instances.